dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -390,7 +390,7 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
 
             Optional<Registry> registry = determineImageRegistry(image);
             Registry registryFound = registry.isEmpty() ? null : registry.get();
-            if (registry.isEmpty() || registryFound == Registry.AMAZON_ECR || registryFound == Registry.GITLAB) {
+            if (registry == null || registryFound == Registry.AMAZON_ECR || registryFound == Registry.GITLAB) {
```

Comment:
Sorry, I think this is wrong; `registry` will never be null. I meant have `registryFound == null`, so that you're using the same variable throughout the if statement. Sorry for the confusion.